cli/command/image/build.go

```diff
@@ -226,6 +184,12 @@ func runBuild(dockerCli command.Cli, options buildOptions) error {
 		remote        string
 	)
 
+	if !options.quiet && dockerCli.ServerInfo().OSType != "windows" {
+		_, _ = fmt.Fprint(dockerCli.Err(), `DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
+
+`)
```

Comment:
Actually; so when is this code hit instead of the code in `processAliases()`? Looks like that would also be hit; perhaps we can simplify and just use a single location.